src/main/java/generationExamples/ExamplesFromFileGenerationImpl.java

```diff
@@ -0,0 +1,72 @@
+package generationExamples;
+
+import main.Configurations;
+import main.Main;
+
+import java.io.BufferedReader;
+import java.io.FileReader;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.time.LocalDateTime;
+import java.time.format.DateTimeFormatter;
+import java.util.List;
+
+public class ExamplesFromFileGenerationImpl implements EquationPattern {
+    private final static String SPLITTER = " ";
+
+    @Override
+    public Integer[][] generationExamples() {
+        BufferedReader reader;
+        int counterExamples = 0;
+        try {
+            reader = new BufferedReader(new FileReader(Configurations.getPathToOpenFile()));
```

Comment:
Please read this part of the article really closely.
http://tutorials.jenkov.com/java-io/bufferedreader.html#closing-a-bufferedreader
and also please take a look at 
https://www.baeldung.com/java-try-with-resources

In short, if you are opening resources then you are responsible for releasing those  resource.